lib/loader/serialization/type.cpp

```diff
@@ -0,0 +1,241 @@
+#include "ast/module.h"
+#include "common/errinfo.h"
+#include "common/log.h"
+#include "loader/loader.h"
+
+namespace WasmEdge {
+namespace Serialize {
+
+std::vector<uint8_t> EncodeU32(uint32_t d) {
+  std::vector<uint8_t> encoded;
+  // unsigned LEB128 encoding
+  do {
+    auto x = d & 0b01111111;
+    d >>= 7;
+    if (d)
+      x |= 0b10000000;
+    encoded.push_back(x);
+  } while (d);
+  return encoded;
+}
+
+std::vector<uint8_t> serializeFunctionType(AST::FunctionType &FuncType) {
+
+  std::vector<uint8_t> result;
+  std::vector<uint8_t> paramTypesSize =
+      EncodeU32(FuncType.getParamTypes.size());
+  std::vector<uint8_t> ReturnTypesSize =
+      EncodeU32(FuncType.getReturnTypes.size());
+  result.push_back(0x60U);
+  result.insert(result.end(), paramTypesSize.begin(), paramTypesSize.end());
+  result.insert(result.end(), FuncType.getParamTypes().begin(),
+                FuncType.getParamTypes().end());
+  result.insert(result.end(), ReturnTypesSize.begin(), ReturnTypesSize.end());
+  result.insert(result.end(), FuncType.getReturnTypes().begin(),
+                FuncType.getReturnTypes().end());
+
+  return result;
+}
+
+std::vector<uint8_t> serializeImportDesc(const AST::ImportDesc &ImpDesc) {

```

Comment:
Separate the serializing of description and type into files.

Reply:
Separted